File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task59_PERPLEXITY_llama-3-sonar-large-32k-chat.java
public class Accumulator {
    public static String accum(String s) {
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            result.append(Character.toUpperCase(s.charAt(i)));
            for (int j = 0; j < i; j++) {
                result.append(Character.toLowerCase(s.charAt(j)));
            }
            if (i < s.length() - 1) {
                result.append("-");
            }
        }
        return result.toString();
    }
    public static void main(String[] args) {
        System.out.println(accum("abcd"));
        System.out.println(accum("RqaEzty"));
        System.out.println(accum("cwAt"));
    }
}

File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task59_PERPLEXITY_llama-3-sonar-large-32k-chat.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

void accum(char *s) {
    for (int i = 0; i < strlen(s); i++) {
        printf("%c", toupper(s[i]));
        for (int j = 0; j < i; j++) {
            printf("%c", tolower(s[j]));
        }
        if (i < strlen(s) - 1) {
            printf("-");
        }
    }
    printf("
");
}

int main() {
    char s1[] = "abcd";
    char s2[] = "RqaEzty";
    char s3[] = "cwAt";
    accum(s1);
    accum(s2);
    accum(s3);
    return 0;
}